x-pack/plugin/core/src/main/resources/ecs-dynamic-mappings.json

```diff
@@ -0,0 +1,198 @@
+{
+  "template": {
+    "mappings": {
+      "date_detection": false,
+      "dynamic_templates": [
+        {
+          "ecs_timestamp": {
+            "mapping": {
+              "ignore_malformed": false,
+              "type": "date"
+            },
+            "match": "@timestamp"
+          }
+        },
+        {
+          "ecs_message_match_only_text": {
+            "mapping": {
+              "type": "match_only_text"
+            },
+            "match": "message"
```

Comment:
Does this also match `error.message` if the [`subobjects: false`](https://www.elastic.co/guide/en/elasticsearch/reference/8.3/subobjects.html) is set at the root level?

If not, we should change to `"path_match": ["message", "*.message"]` and also adjust other places where we're using `match`.

Reply:
It starts to feel like we need the `subobjects: false` either within this PR or immediately after.
Anyway, the `error.message` mapping is already [tested](https://github.com/elastic/elasticsearch/pull/96171/files#diff-c5877225937baee0eb47d89e6502503d5813b1df3048a6a8e1c48c0f2a6a2855R333), so even if we follow up with `subobjects: false` on a different PR, it should either verify it works or force us to fix.

Reply:
> It starts to feel like we need the subobjects: false either within this PR or immediately after.

I'm good with either. If we merge this PR without optimizations for subobjects: false, we should create a direct follow-up PR, however. I think it makes sense to have that change in the commit history, which speaks for doing a follow-up PR.